From 660101e3cdd2f8e8bc627517691dfb885a5c8302 Mon Sep 17 00:00:00 2001
From: Iustin Ventaniuc <iustiniv@google.com>
Date: Thu, 13 Nov 2025 09:48:16 +0000
Subject: [PATCH] Handle loadDescription OutOfMemoryError in DeviceAdminInfo

loadDescription was potentially vulnerable to an attack which causes a
DoS exploit by injecting a maliciously large string into the Receiver's
label.

Bug: 443062265
Test: manual
Flag: EXEMPT BUGFIX
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:06a5b2327caa3aa8843496458e98b9bb070df6e5
Merged-In: Icab26c4b77e73f0fcb9a560e3211482ebe2f37bf
Change-Id: Icab26c4b77e73f0fcb9a560e3211482ebe2f37bf
---
 core/java/android/app/admin/DeviceAdminInfo.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/java/android/app/admin/DeviceAdminInfo.java b/core/java/android/app/admin/DeviceAdminInfo.java
index 7b46db16f80a..7616d805c31c 100644
--- a/core/java/android/app/admin/DeviceAdminInfo.java
+++ b/core/java/android/app/admin/DeviceAdminInfo.java
@@ -473,8 +473,12 @@ public final class DeviceAdminInfo implements Parcelable {
      */
     public CharSequence loadDescription(PackageManager pm) throws NotFoundException {
         if (mActivityInfo.descriptionRes != 0) {
-            return pm.getText(mActivityInfo.packageName,
+            try {
+                return pm.getText(mActivityInfo.packageName,
                     mActivityInfo.descriptionRes, mActivityInfo.applicationInfo);
+            } catch (OutOfMemoryError e) {
+                throw new NotFoundException();
+            }
         }
         throw new NotFoundException();
     }
-- 
2.53.0.335.g19a08e0c02-goog

